dune/xt/functions/interfaces/flux-function.hh

```diff
@@ -0,0 +1,122 @@
+//          with "runtime exception" (http://www.dune-project.org/license.html)
+// Authors:
+//   Felix Schindler (2017 - 2018)
+//   René Fritze     (2018)
+//   Tim Keil        (2018)
+//   Tobias Leibner  (2017)
+
+#ifndef DUNE_XT_FUNCTIONS_INTERFACES_FLUX_FUNCTION_HH
+#define DUNE_XT_FUNCTIONS_INTERFACES_FLUX_FUNCTION_HH
+
+#include <memory>
+#include <ostream>
+#include <string>
+#include <type_traits>
+#include <vector>
+
+#include <dune/common/parallel/mpihelper.hh>
+
+#include <dune/common/deprecated.hh>
+
+#include <dune/grid/io/file/vtk.hh>
+
+#include <dune/xt/common/filesystem.hh>
+#include <dune/xt/common/memory.hh>
+#include <dune/xt/common/type_traits.hh>
+#include <dune/xt/grid/dd/subdomains/grid.hh>
+#include <dune/xt/grid/grids.hh>
+#include <dune/xt/grid/layers.hh>
+#include <dune/xt/grid/view/from-part.hh>
+#include <dune/xt/grid/type_traits.hh>
+
+#include <dune/xt/functions/exceptions.hh>
+#include <dune/xt/functions/type_traits.hh>
+#include <dune/xt/functions/base/visualization.hh>
+
+#include "element-flux-functions.hh"
+
+namespace Dune {
+namespace XT {
+namespace Functions {
+
+
+/**
+ * \brief Interface for functions f(x, u) which can be partially localized to an element in the first variable x.
+ */
+template <class Element, size_t stateDim = 1, size_t rangeDim = 1, size_t rangeDimCols = 1, class RangeField = double>
+class FluxFunctionInterface : public Common::ParametricInterface
+{
+  using ThisType = FluxFunctionInterface;
+
+public:
+  using LocalFunctionType = ElementFluxFunctionInterface<Element, stateDim, rangeDim, rangeDimCols, RangeField>;
+
+  using ElementType = typename LocalFunctionType::ElementType;
+  using DomainFieldType = typename LocalFunctionType::DomainFieldType;
+  using StateType = typename LocalFunctionType::StateType;
+  using DomainType = typename LocalFunctionType::DomainType;
+  static const constexpr size_t domain_dim = LocalFunctionType::domain_dim;
+  static const constexpr size_t state_dim = LocalFunctionType::state_dim;
+  using RangeFieldType = typename LocalFunctionType::RangeFieldType;
+  static const constexpr size_t range_dim = LocalFunctionType::range_dim;
+  static const constexpr size_t range_dim_cols = LocalFunctionType::range_dim_cols;
+
+  using E = typename LocalFunctionType::E;
+  using D = typename LocalFunctionType::D;
+  static const constexpr size_t d = LocalFunctionType::d;
+  static const constexpr size_t s = LocalFunctionType::s;
+  using R = typename LocalFunctionType::R;
+  using S = typename LocalFunctionType::S;
+  static const constexpr size_t r = LocalFunctionType::r;
+  static const constexpr size_t rC = LocalFunctionType::rC;
+
+  static const constexpr bool available = false;
+
+  FluxFunctionInterface(const Common::ParameterType& param_type = {})
+    : Common::ParametricInterface(param_type)
+  {}
+
+  virtual ~FluxFunctionInterface() = default;
+
+  static std::string static_id()
+  {
+    return "dune.xt.functions.gridfunction";
```

Comment:
I would drop the static_id completely, we currently have no use for it ...

Reply:
Done.